pkg/accounts/kubernetes/validator.go

```diff
@@ -32,21 +36,49 @@ type kubernetesAccountValidator struct {
 }
 
 func (k *kubernetesAccountValidator) Validate(spinSvc v1alpha2.SpinnakerServiceInterface, c client.Client, ctx context.Context, log logr.Logger) error {
-	if err := k.validateSettings(ctx, log); err != nil {
+	err := k.validateSettings(ctx, log)
+	if err != nil {
 		return err
 	}
-	if k.account.Auth != nil && k.account.Auth.UseServiceAccount && spinSvc == nil {
-		// don't validate if the spinnaker account needs a k8s service account to run validations, and there's no SpinnakerService yet
-		return nil
+	if k.account.Auth != nil && k.account.Auth.UseServiceAccount {
+		spinSvc, err = k.ensureSpinSvc(spinSvc, c, ctx)
+		if err != nil {
+			return err
+		}
+		if spinSvc == nil {
+			// don't validate if the spinnaker account needs a k8s service account to run validations, and there's no SpinnakerService yet
+			return nil
+		}
 	}
-	config, err := k.makeClient(ctx, spinSvc)
+	config, err := k.makeClient(ctx, spinSvc, c)
 	if err != nil {
 		return err
 	}
 	return k.validateAccess(config)
 }
 
-func (k *kubernetesAccountValidator) makeClient(ctx context.Context, spinSvc v1alpha2.SpinnakerServiceInterface) (clientcmd.ClientConfig, error) {
+func (k *kubernetesAccountValidator) ensureSpinSvc(spinSvc v1alpha2.SpinnakerServiceInterface, c client.Client, ctx context.Context) (v1alpha2.SpinnakerServiceInterface, error) {
+	if spinSvc != nil {
+		return spinSvc, nil
+	}
+	i := SpinnakerServiceBuilder.NewList()
+	sc, err := secrets.FromContextWithError(ctx)
+	if err != nil {
+		return nil, err
+	}
+	list, err := util.GetSpinnakerServices(i, sc.Namespace, c)
+	if err != nil {
+		return nil, err
+	}
+	if len(list) == 0 {
+		return nil, nil
+	} else {
+		// there's only one spinnaker service per namespace
```

Comment:
There **should** be one spinnaker if validation was on. I'm fine taking the first one though!